clock/scheduledJobs/queueStatsActivityLogs.js

```diff
@@ -0,0 +1,241 @@
+let queueTimeArray = [];
+
+// Create Redis connection
+const client = redis.createClient(rconf.instanceUrl.realtimeLogging);
+
+/**
+ * Convert array to string
+ *
+ * @param {array} array - array of integers
+ * @returns {string} string - string of array
+ */
+function arrayToString(array) {
+  return array.join(',');
+}
+
+/**
+ * Convert array to string
+ *
+ * @param {string} string - string of array
+ * @returns {array} array - array of integers
+ */
+function stringToArray(string) {
+  if (string.length) {
+    const array = string.split(',');
+    for (let i = 0; i < array.length; i++) {
+      array[i] = Number(array[i]);
+    }
+
+    return array;
+  }
+
+  return [];
+}
+
+/**
+ * Convert string to Array then add element and again conver
+ * to string
+ *
+ * @param {string} string - string of array
+ * @param {integer} qt - queueTime
+ * @returns {string} string - string of array
+ */
+function addToArray(string, qt) {
+  const array = stringToArray(string);
+  array.push(qt);
+  return arrayToString(array);
+}
+
+/**
+ * Calculate timestamp i.e 2017.04.11.15.58.
+ *
+ * @returns {string} timestamp - timestamp string
+ */
+function createTimeStamp() {
+  const date = new Date();
+  return date.getUTCFullYear() + '.' + (date.getUTCMonth() + ONE) + '.' +
+    date.getUTCDate() + '.' + date.getUTCHours() + '.' + date.getUTCMinutes();
+}
+
+/**
+ * Calculate Statistics based on queue timing array.
+ *
+ * @param {array} qt - array of queue timings
+ * @returns {object} stats - statistics based on queue timing array
+ *     i.e median, average and 95th Percentile
+ */
+function calculateStats(qt) {
+  let median = 0;
+  let average = 0;
+  let percentile95th = 0;
+  const stats = {};
+  const length = qt.length;
+
+  // sort the queue time array
+  qt.sort((a, b) => a - b);
+
+  // calculate sum of queue time
+  const sum = qt.reduce((a, b) => a + b, ZERO);
+
+  // calculate middle of array
+  const middle = Math.floor(length / TWO);
+  average = sum / length;
+
+  // calculate median
+  if (middle % TWO) {
+    median = (qt[middle - ONE] + qt[middle]) / FLOAT_TWO;
+  } else {
+    median = qt[middle];
+  }
+
+  // calculate 95th Percentile
+  const index = Math.round(PERCENTILE_95TH * length);
+  percentile95th = qt[index - ONE];
+
+  // construct stats object
+  stats.averageQueueTimeMillis = average.toFixed(TWO) + 'ms';
+  stats.medianQueueTimeMillis = median.toFixed(TWO) + 'ms';
+  stats.queueTimeMillis95th = percentile95th + 'ms';
+  stats.queueTimeMillis95thNumber = percentile95th;
+
+  return stats;
+}
+
+/**
+ * Construct log object
+ *
+ * @param {object} qStats - queue time object from redis
+ * @returns {object} queueStats - queue time stats log object
+ */
+function constructLogObject(qStats) {
+  const queueStats = {};
+
+  queueStats.jobCount = qStats.jobCount;
+  queueStats.recordCount = qStats.recordCount;
+  queueTimeArray = stringToArray(qStats.queueTimeArray);
+  queueStats.minute = qStats.minute;
+
```

Comment:
@harshkothari410 @iamigo Just looked at the logs in papertrail. It begins "minute=2017.4.13.20.17...." . Should this be timestamp?